rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/aws/lambda/TbLambdaNodeTest.java

```diff
@@ -0,0 +1,302 @@
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.rule.engine.api.TbContext;
+import org.thingsboard.rule.engine.api.TbNodeConfiguration;
+import org.thingsboard.rule.engine.api.TbNodeException;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.msg.TbMsgType;
+import org.thingsboard.server.common.msg.TbMsg;
+import org.thingsboard.server.common.msg.TbMsgMetaData;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class TbLambdaNodeTest {
+
+    private final DeviceId DEVICE_ID = new DeviceId(UUID.fromString("ddb88645-7379-4a08-a51c-e84a0b4b3d88"));
+    
+    private TbLambdaNode node;
+    private TbLambdaNodeConfiguration config;
+
+    @Mock
+    private TbContext ctx;
+    @Mock
+    private AWSLambdaAsync clientMock;
+
+    @BeforeEach
+    public void setUp() {
+        node = new TbLambdaNode();
+        config = new TbLambdaNodeConfiguration().defaultConfiguration();
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    @ValueSource(strings = "  ")
+    public void givenInvalidFunctionName_whenInit_thenThrowsException(String funcName) {
+        config.setQualifier(funcName);
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        assertThatThrownBy(() -> node.init(ctx, configuration))
+                .isInstanceOf(TbNodeException.class)
+                .hasMessage("Invalid function name");
+    }
+
+    @Test
+    void givenInvocationTypeIsNull_whenInit_thenThrowsException() {
+        config.setFunctionName("new-function");
+        config.setInvocationType(null);
+        var configuration = new TbNodeConfiguration(JacksonUtil.valueToTree(config));
+        assertThatThrownBy(() -> node.init(ctx, configuration))
+                .isInstanceOf(TbNodeException.class)
+                .hasMessage("Invocation type must be set!");
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void givenFunctionNameAndQualifierPatterns_whenOnMsg_thenTellSuccess(String functionName, String qualifier)
+            throws TbNodeException, ExecutionException, InterruptedException {
+        init();
```

Comment:
the init method should have all the parameters set by default. If you need to change some parameter after that you can always use setters.